### PR TITLE
Make init-tools.sh work against a Windows drive mounted on Linux

### PR DESCRIPTION
**Pull request by danmosemsft**
So far the only issue I've hit is that the 3 files we read in for literal values will end in \r\n, causing the environment variables to end with \r. This fixes that.

### CODE REVIEW

**Comment by danmosemsft** [comment]
Incidentally the diagnosability of init-tools.sh could be improved. It puts half its output in init-tools.log and half to the console, and does not log all its commands (such as the curl command).

File: init-tools.sh [hunk 1]
```diff
@@ -8,9 +8,9 @@ __DOTNET_PATH=$__TOOLRUNTIME_DIR/dotnetcli
 __DOTNET_CMD=$__DOTNET_PATH/dotnet
 if [ -z "$__BUILDTOOLS_SOURCE" ]; then __BUILDTOOLS_SOURCE=https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json; fi
 export __BUILDTOOLS_USE_CSPROJ=true
-__BUILD_TOOLS_PACKAGE_VERSION=$(cat $__scriptpath/BuildToolsVersion.txt)
-__DOTNET_TOOLS_VERSION=$(cat $__scriptpath/DotnetCLIVersion.txt)
-__ILASM_VERSION=$(cat $__scriptpath/tools-local/ILAsmVersion.txt)
+__BUILD_TOOLS_PACKAGE_VERSION=$(cat $__scriptpath/BuildToolsVersion.txt | sed 's/\r$//') # remove CR if mounted repo on Windows drive
+__DOTNET_TOOLS_VERSION=$(cat $__scriptpath/DotnetCLIVersion.txt | sed 's/\r$//') # remove CR if mounted repo on Windows drive
+__ILASM_VERSION=$(cat $__scriptpath/tools-local/ILAsmVersion.txt | sed 's/\r$//') # remove CR if mounted repo on Windows drive
```
Comment (dagood) [inline init-tools.sh]: Seems like a good place to use a function.

**Comment by dagood** [comment]
Yeah, one reason init-tools.sh doesn't see many improvements is that currently there's no way to improve all the repos' init-tools scripts at the same time--they get out of sync and become hard to update in a way that can be applied to all.

I have a lot of work done on a build script auto-updater (https://github.com/dotnet/buildtools/issues/1709) but I haven't had time recently to fix the last few issues I had with it.

**Comment by weshaggard** [comment]
FYIW I used to map windows shares in my linux docker containers but there are lots of issues with doing it these newlines are just the tip of the iceberg. You will have crazy performance issues as it is much slower to build and you will hit issues with tar not liking it. I recommend having a unique clone just for your unix docker containers.

File: init-tools.sh (hunk 1 above)
Comment (danmosemsft) [reply init-tools.sh]: @dagood mind if I leave it like that? I think it is easier to read/self explanatory .. a function would only save a few characters.

**Comment by danmosemsft** [comment]
@weshaggard I'm using it from my Ubuntu on Windows shell: maybe the performance is not so bad. I'll find out.

File: init-tools.sh (hunk 1 above)
Comment (dagood) [reply init-tools.sh]: Fine to me to leave this for now, but I still prefer a function for maintainability, especially if/when we add more file reading. I also wouldn't think it it counts as self-explanatory if it has a comment on every line it's done.

**Pull request merged**